### PR TITLE
Update `azurerm_policy_set_definition` - Support `policy_definition_group`

### PR DESCRIPTION
**Pull request by ArcturusZhang**

### CODE REVIEW

**Comment by ArcturusZhang** [comment]
Tests all passed!
image

File: azurerm/internal/services/policy/policy_set_definition_data_source.go [hunk 1]
```diff
@@ -89,6 +89,14 @@ func dataSourceArmPolicySetDefinition() *schema.Resource {
 							Type:     schema.TypeString,
 							Computed: true,
 						},
+
+						"group_names": {
```
Comment (katbyte) [inline azurerm/internal/services/policy/policy_set_definition_data_source.go:93]: should this be
```suggestion
						"policy_group_names": {
```

File: azurerm/internal/services/policy/policy_set_definition_data_source.go [hunk 2]
```diff
@@ -97,6 +105,39 @@ func dataSourceArmPolicySetDefinition() *schema.Resource {
...
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+
+						"description": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+
+						"additional_metadata_id": {
```
Comment (katbyte) [inline azurerm/internal/services/policy/policy_set_definition_data_source.go:134]: should this be
```suggestion
						"additional_metadata_resource_id": {
```

File: azurerm/internal/services/policy/policy_set_definition_resource.go [hunk 3]
```diff
@@ -152,9 +154,57 @@ func resourceArmPolicySetDefinition() *schema.Resource {
 							Optional: true,
 							Computed: true,
 						},
+
+						"group_names": {
```
Comment (katbyte) [inline azurerm/internal/services/policy/policy_set_definition_resource.go:158]: same as above

File: azurerm/internal/services/policy/policy_set_definition_resource.go [hunk 4]
```diff
@@ -152,9 +154,57 @@ func resourceArmPolicySetDefinition() *schema.Resource {
...
+							ValidateFunc: validation.StringIsNotEmpty,
+						},
+
+						"description": {
+							Type:         schema.TypeString,
+							Optional:     true,
+							ValidateFunc: validation.StringIsNotEmpty,
+						},
+
+						"additional_metadata_id": {
```
Comment (katbyte) [inline azurerm/internal/services/policy/policy_set_definition_resource.go:199]: same as above

File: azurerm/internal/services/policy/tests/policy_set_definition_resource_test.go [hunk 5]
```diff
@@ -806,6 +825,54 @@ VALUES
...
+	{
+      "allowedLocations": {"value": "[parameters('allowedLocations')]"}
+    }
+VALUES
+    group_names          = ["group-1", "group-2"]
+  }
+
+  policy_definition_group {
+    name = "redundant"
+  }
```
Comment (katbyte) [inline azurerm/internal/services/policy/tests/policy_set_definition_resource_test.go:863]: could we add another test that sets all the properties on the resource? and then go from this config -> all -> back?

**Review by katbyte — CHANGES_REQUESTED** [review]
Thanks @ArcturusZhang - overall this looks great, however I've got some questions about some of the property names and we should add a test that sets all the new properties.

File: azurerm/internal/services/policy/policy_set_definition_data_source.go (hunk 1 above)
Comment (ArcturusZhang) [reply azurerm/internal/services/policy/policy_set_definition_data_source.go]: Done

File: azurerm/internal/services/policy/policy_set_definition_data_source.go (hunk 2 above)
Comment (ArcturusZhang) [reply azurerm/internal/services/policy/policy_set_definition_data_source.go]: Done

File: azurerm/internal/services/policy/policy_set_definition_resource.go (hunk 3 above)
Comment (ArcturusZhang) [reply azurerm/internal/services/policy/policy_set_definition_resource.go]: Done

File: azurerm/internal/services/policy/policy_set_definition_resource.go (hunk 4 above)
Comment (ArcturusZhang) [reply azurerm/internal/services/policy/policy_set_definition_resource.go]: Done

File: azurerm/internal/services/policy/tests/policy_set_definition_resource_test.go [hunk 6]
```diff
@@ -806,6 +839,108 @@ VALUES
...
+VALUES
+    policy_group_names   = ["group-1", "group-2"]
+  }
+
+  policy_definition_group {
+    name = "redundant"
+  }
+
+  policy_definition_group {
+    name         = "group-1"
```
Comment (katbyte) [inline azurerm/internal/services/policy/tests/policy_set_definition_resource_test.go:928]: Could we have a case where it contains capitals?
```suggestion
    name         = "Group-1"
```

as the validation allows themn

File: azurerm/internal/services/policy/tests/policy_set_definition_resource_test.go [hunk 7]
```diff
@@ -806,6 +839,108 @@ VALUES
...
+    display_name = "group-display-1"
```
Comment (katbyte) [inline azurerm/internal/services/policy/tests/policy_set_definition_resource_test.go:929]: Same here
```suggestion
    display_name = "Group-Display-1"
```

**Review by katbyte — CHANGES_REQUESTED** [review]
Thanks @ArcturusZhang - just a couple changes to the tests and this should be g2g

File: azurerm/internal/services/policy/tests/policy_set_definition_resource_test.go (hunk 6 above)
Comment (ArcturusZhang) [reply azurerm/internal/services/policy/tests/policy_set_definition_resource_test.go:928]: No Problem

File: azurerm/internal/services/policy/tests/policy_set_definition_resource_test.go (hunk 7 above)
Comment (ArcturusZhang) [reply azurerm/internal/services/policy/tests/policy_set_definition_resource_test.go:929]: No problem

**Comment by ArcturusZhang** [comment]
Hi @katbyte I have resolved the comments, please have a look, thank you!

**Review by katbyte — APPROVED** [review]
Thanks @ArcturusZhang - this LGTM 👍

**Pull request merged**